lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -37,6 +37,57 @@ public static void tearDownAfterClass() throws Exception {
 	}
 
 	/**
+	 * Connect to the service and attach a channel.
+	 * Don't publish any messages; disconnect and then reconnect; verify that
+	 * the channel is still attached.
+	 */
+	@Test
+	public void resume_none() {
+		AblyRealtime ably = null;
+		String channelName = "resume_none";
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+			ably = new AblyRealtime(opts);
+
+			/* create and attach channel */
+			final Channel channel = ably.channels.get(channelName);
+			System.out.println("Attaching");
+			channel.attach();
+			(new ChannelWaiter(channel)).waitFor(ChannelState.attached);
+			assertEquals("Verify attached state reached", channel.state, ChannelState.attached);
+
+			/* disconnect the connection, without closing;
+			 * NOTE this depends on knowledge of the internal structure
+			 * of the library, to simulate a dropped transport without
+			 * causing the connection itself to be disposed */
+			System.out.println("Simulating dropped transport");
+			ConnectionWaiter connectionWaiter = new ConnectionWaiter(ably.connection);
+			ably.connection.connectionManager.requestState(ConnectionState.disconnected);
+			connectionWaiter.waitFor(ConnectionState.disconnected);
+
+			/* Wait for the connection to reconnect by itself. */
+			System.out.println("Waiting for reconnection");
+			connectionWaiter.waitFor(ConnectionState.connected);
+			assertEquals("Verify connected state is reached", ConnectionState.connected, ably.connection.state);
+
+			/* wait */
+			System.out.println("Got reconnection; waiting 2s");
+			try { Thread.sleep(2000L); } catch(InterruptedException e) {}
+
+			/* Check the channel is still attached. */
+			assertEquals("Verify channel still attached", channel.state, ChannelState.attached);
```

Comment:
Would be nice to publish a message over REST to that channel to ensure that it really is active as opposed to the client library thinking it's active.